File: worblehat-web/src/test/java/de/codecentric/psd/worblehat/web/formdata/ReturnAllBooksFormDataTest.java
package de.codecentric.psd.worblehat.web.formdata;

import org.junit.Before;
import org.junit.Test;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.*;

public class ReturnAllBooksFormDataTest {

    private ReturnAllBooksFormData formData;

    @Before
    public void setUp() throws Exception {
        formData = new ReturnAllBooksFormData();
    }

    @Test
    public void getTitleWithWhitespaces() {
        formData.setTitle(" mit ");
        assertThat(formData.getTitle(), is("mit"));
    }

    @Test
    public void getTitleWithoutWhitespaces() {
        formData.setTitle("ohne");
        assertThat(formData.getTitle(), is("ohne"));
    }

    @Test
    public void getEmailAddressWithWhitespaces() {
        formData.setEmailAddress(" mit ");
        assertThat(formData.getEmailAddress(), is("mit"));
    }

    @Test
    public void getEmailAddressWithoutWhitespaces() {
        formData.setEmailAddress("ohne");
        assertThat(formData.getEmailAddress(), is("ohne"));
    }

    @Test
    public void getIsbnWithWhitespaces() {
        formData.setIsbn(" mit ");
        assertThat(formData.getIsbn(), is("mit"));
    }

    @Test
    public void getIsbnWithoutWhitespaces() {
        formData.setIsbn("ohne");
        assertThat(formData.getIsbn(), is("ohne"));
    }

    @Test
    public void getRadioButtonValues() {
        formData.setRadioButtonValues(new String[]{"Test1", "Test2"});
        assertThat(formData.getRadioButtonValues(), is(new String[]{"Test1", "Test2"}));
    }

    @Test
    public void getRadioButtonSelectionWithWhitespaces() {
        formData.setRadioButtonSelection(" mit ");
        assertThat(formData.getRadioButtonSelection(), is("mit"));
    }

    @Test
    public void getRadioButtonSelectionWithoutWhitespaces() {
        formData.setRadioButtonSelection("ohne");
        assertThat(formData.getRadioButtonSelection(), is("ohne"));
    }

}

File: worblehat-web/src/test/java/de/codecentric/psd/worblehat/web/formdata/ReturnAllBooksFormDataTest.kt
package de.codecentric.psd.worblehat.web.formdata

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat
import org.junit.Before
import org.junit.Test

class ReturnAllBooksFormDataTest {

    private var formData: ReturnAllBooksFormData? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        formData = ReturnAllBooksFormData()
    }

    @Test
    fun getTitleWithWhitespaces() {
        formData!!.title = " mit "
        assertThat(formData!!.title, `is`("mit"))
    }

    @Test
    fun getTitleWithoutWhitespaces() {
        formData!!.title = "ohne"
        assertThat(formData!!.title, `is`("ohne"))
    }

    @Test
    fun getEmailAddressWithWhitespaces() {
        formData!!.emailAddress = " mit "
        assertThat(formData!!.emailAddress, `is`("mit"))
    }

    @Test
    fun getEmailAddressWithoutWhitespaces() {
        formData!!.emailAddress = "ohne"
        assertThat(formData!!.emailAddress, `is`("ohne"))
    }

    @Test
    fun getIsbnWithWhitespaces() {
        formData!!.isbn = " mit "
        assertThat(formData!!.isbn, `is`("mit"))
    }

    @Test
    fun getIsbnWithoutWhitespaces() {
        formData!!.isbn = "ohne"
        assertThat(formData!!.isbn, `is`("ohne"))
    }

    @Test
    fun getRadioButtonValues() {
        formData!!.radioButtonValues = arrayOf("Test1", "Test2")
        assertThat(formData!!.radioButtonValues, `is`(arrayOf("Test1", "Test2")))
    }

    @Test
    fun getRadioButtonSelectionWithWhitespaces() {
        formData!!.radioButtonSelection = " mit "
        assertThat(formData!!.radioButtonSelection, `is`("mit"))
    }

    @Test
    fun getRadioButtonSelectionWithoutWhitespaces() {
        formData!!.radioButtonSelection = "ohne"
        assertThat(formData!!.radioButtonSelection, `is`("ohne"))
    }

}